datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -643,6 +646,8 @@ private void handleAssignmentChange(boolean isDatastreamUpdate) throws TimeoutEx
         .stream()
         .flatMap(Collection::stream)
         .collect(Collectors.toMap(DatastreamTask::getDatastreamTaskName, Function.identity())));
+    _assignedDatastreamTasks.keySet().removeAll(_failedDatastreamTasks.keySet());
```

Comment:
what if this code does not get executed because of some exception. and in the next pass the task initialization passes. You will remove the entry from the `_assignedDatastreamTasks` list.

How long do you want to persist the list of `_failedDatastreamTasks`? Plz add it in the object description, so that we can evaluate that this object state is cleared after that.